src/components/HeaderForm/HeaderForm.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+
+interface Props {
+  disabledInput: boolean;
+  handleSubmitForm: (event: React.FormEvent<HTMLFormElement>) => void;
+  handleQueryChange: (query: string) => void;
+}
+
+export const HeaderForm: React.FC<Props> = ({
+  disabledInput, handleSubmitForm, handleQueryChange,
+}) => {
+  const [query, setQuery] = useState('');
```

Comment:
why do you need to store the `query `in two components? You already have the `query `state in the App. Always use one source of truth in React